src/SW_netCDF_Input.c

```diff
@@ -0,0 +1,3540 @@
+                }
+            }
+        }
+    }
+}
+
+/*
+@brief Read input netCDF variables that the user will provide
+
+@param[in,out] SW_netCDFIn Constant netCDF input file information
+@param[in] SW_PathInputs Struct of type SW_PATH_INPUTS which
+holds basic information about input files and values
+@param[in] startYr End year of the simulation
+@param[in] endYr End year of the simulation
+@param[out] LogInfo Holds information on warnings and errors
+*/
+void SW_NCIN_read_input_vars(
+    SW_NETCDF_IN *SW_netCDFIn,
+    SW_PATH_INPUTS *SW_PathInputs,
+    TimeInt startYr,
+    TimeInt endYr,
+    LOG_INFO *LogInfo
+) {
+
+    FILE *f;
+    int lineno = 0;
+    char *MyFileName;
+    int scanRes;
+    int infoIndex;
+    int copyInfoIndex = 0;
+    int index;
+    int strInfoInd;
+    char inbuf[LARGE_VALUE] = {'\0'};
+    char input[NIN_VAR_INPUTS][MAX_ATTVAL_SIZE] = {"\0"};
+    const char *readLineFormat =
+        "%10[^\t]\t%50[^\t]\t%20[^\t]\t%1[^\t]\t%255[^\t]\t%50[^\t]\t"
+        "%255[^\t]\t%255[^\t]\t%255[^\t]\t%255[^\t]\t%255[^\t]\t%255[^\t]\t"
+        "%255[^\t]\t%255[^\t]\t%255[^\t]\t%255[^\t]\t%255[^\t]\t%255[^\t]\t"
+        "%255[^\t]\t%255[^\t]";
+
+    /* Locally handle the weather stride information where -2 is
+       the default value so we can tell that the value hasn't been set yet
+
+       0: number of years within a stride
+       1: stride start year */
+    int inWeathStrideInfo[2] = {-2, -2};
+    int tempStrideInfo[2];
+
+    const int keyInd = 0;
+    const int SWVarNameInd = 1;
+    const int SWUnitInd = 2;
+    const int doInputInd = 3;
+    const int ncFileNameInd = 4;
+    const int ncVarNameInd = 5;
+    const int ncVarUnitsInd = 6;
+    const int ncGridTypeInd = 7;
+    const int ncCRSNameInd = 8;
+    const int ncCRSEquivInd = 9;
+    const int ncXAxisInd = 10;
+    const int ncYAxisInd = 11;
+    const int ncZAxisInd = 12;
+    const int ncTAxisInd = 13;
+    const int ncStYrInd = 14;
+    const int ncStStartInd = 15;
+    const int ncStPatInd = 16;
+    const int ncCalendarInd = 17;
+    const int ncVAxisInd = 18;
+    const int userComInd = 19;
+
+    int inKey = -1;
+    int inVarNum = -1;
+    int doInput = 0;
+    int maxVarIter = 1;
+    int varIter;
+    const int allVegInc = 5;
+
+    Bool copyInfo = swFALSE;
+    Bool isIndexFile = swFALSE;
+    Bool isAllVegVar = swFALSE;
+
+    /* Acceptable non-numerical values for stride year and stride start,
+       respectively */
+    const char *accStrVal[] = {"Inf", "NA"};
+    char **varInfoPtr = NULL;
+
+    MyFileName = SW_PathInputs->txtInFiles[eNCIn];
+    f = OpenFile(MyFileName, "r", LogInfo);
+    if (LogInfo->stopRun) {
+        return; /* Exit function prematurely due to error */
+    }
+
+    SW_NCIN_alloc_input_var_info(SW_netCDFIn, LogInfo);
+    if (LogInfo->stopRun) {
+        return; /* Exit function prematurely due to error */
+    }
+
+    /* Ignore the first row (column names) */
+    GetALine(f, inbuf, MAX_FILENAMESIZE);
```

Comment:
Our current code only works for the hard-coded order of columns in input.tsv. If the user changes the column order, the code produces error messages that are not helpful, e.g., "Start year of weather input file for 'tasmax' is <= 0."

I suggest that `SW_NCIN_read_input_vars()` checks that the column names are in the exact order that the code expects and fails otherwise. Similarly, `SW_NCOUT_read_out_vars()` should also check the column names.